workers/worker-performance.worker.js

```diff
@@ -0,0 +1,124 @@
+}, "Can use performance.getEntriesByType in workers");
+
+test(function testPerformanceEntryOrder () {
+    assert_greater_than(marks[0].startTime, start, "First mark startTime is after a time before it");
+    assert_greater_than(marks[1].startTime, marks[0].startTime, "Second mark startTime is after first mark startTime");
+    assert_equals(measures[0].startTime, marks[0].startTime, "measure's startTime is the first mark's startTime");
+}, "Performance marks and measures seem to be working correctly in workers");
+
+test(function testPerformanceClearing () {
+    performance.clearMarks();
+    assert_equals(performance.getEntriesByType("mark").length, 0, "clearMarks cleared the marks");
+    performance.clearMeasures();
+    assert_equals(performance.getEntriesByType("measure").length, 0, "clearMeasures cleared the measures");
+}, "Can use clearMarks and clearMeasures in workers");
+
+test(function testPerformanceResourceTiming () {  // Resource timing
+    var start = performance.now();
+    var xhr = new XMLHttpRequest();
+     // Do a synchronous request and add a little artificial delay
+    xhr.open("GET", "/resources/testharness.js?pipe=trickle(d0.25)", false);
+    xhr.send();
+     // The browser might or might not have added a resource performance entry for the importScripts() above; we're only interested in xmlhttprequest entries
+    var entries = performance.getEntriesByType("resource").filter(entry => entry.initiatorType == "xmlhttprequest");
+    assert_equals(entries.length, 1, "getEntriesByType(\"resource\") returns one entry with initiatorType of xmlhttprequest");
+    assert_true(!!entries[0].name.match(/\/resources\/testharness.js/), "Resource entry has loaded url as its name");
+    assert_equals(entries[0].entryType, "resource", "Resource entry has correct entryType");
+    assert_equals(entries[0].initiatorType, "xmlhttprequest", "Resource entry has correct initiatorType");
+    var currentTimestamp = start;
+    var currentTimestampName = "a time before it";
+    [
+        "startTime", "fetchStart", "requestStart", "responseStart", "responseEnd"
+    ].forEach((name) => {
+        var timestamp = entries[0][name];
+        // We want to skip over values that are 0 because of TAO securty rescritions
+        // Or else this test will fail. This can happen for "requestStart", "responseStart".
+        if (timestamp != 0) {
+            assert_greater_than_equal(timestamp, currentTimestamp, "Resource entry " + name + " is after " + currentTimestampName);
+            currentTimestamp = timestamp;
+            currentTimestampName = name;
+        }
+    });
+    assert_greater_than(entries[0].responseEnd, entries[0].startTime, "The resource request should have taken at least some time");
+    assert_approx_equals(entries[0].responseEnd - entries[0].responseStart, 250, 50, "Resource timing numbers reflect reality somewhat");
```

Comment:
I think we need to relax this requirement? Executing this test against w3c-test.org I'm seeing significantly higher download times, in 450ms+ range. Perhaps change to 250ms+?

![image](https://user-images.githubusercontent.com/10652/30197163-dbd87868-941a-11e7-8c02-52386bfdbead.png)

Test url: https://w3c-test.org/submissions/7213/workers/worker-performance.worker.html

Reply:
Ah, that makes sense.  I was running this test with a local server, but there's no upper bound to how long the request can take with a remote server.